src/net/gateway-archive.ts

```diff
@@ -0,0 +1,90 @@
+import {
+    VOCDONI_MAINNET_ENTITY_ID,VOCDONI_RINKEBY_ENTITY_ID, VOCDONI_GOERLI_ENTITY_ID, VOCDONI_XDAI_ENTITY_ID, VOCDONI_SOKOL_ENTITY_ID, XDAI_ENS_REGISTRY_ADDRESS, XDAI_PROVIDER_URI, XDAI_CHAIN_ID,
+    SOKOL_CHAIN_ID, SOKOL_PROVIDER_URI, SOKOL_ENS_REGISTRY_ADDRESS, XDAI_STG_ENS_REGISTRY_ADDRESS, VOCDONI_XDAI_STG_ENTITY_ID
+} from "../constants"
+import { TextRecordKeys } from "../models/entity"
+import { IGateway } from "./gateway";
+import { DVoteGatewayResponseBody } from "./gateway-dvote";
+import { IGatewayPool } from "./gateway-pool";
+import { keccak256 } from "@ethersproject/keccak256"
+
+export type EthNetworkID = "mainnet" | "rinkeby" | "goerli" | "xdai" | "sokol"
+
+export class GatewayArchive {
+
+    /**
+     * Fetch the process data on the IPFS archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    public static getProcessFromArchive(processId: string, gateway: IGateway | IGatewayPool, errorMessage: string): Promise<DVoteGatewayResponseBody> {
+        return this.getArchiveUri(gateway)
+            .then((archiveUri: ContentUri) => {
+                return this.getArchiveFile(archiveUri, processId, gateway)
+            })
+            .then((result: string) => {
+                return JSON.parse(result)
+            })
+            .catch((error) => {
+                throw new Error(errorMessage)
+            })
+    }
+
+    /**
+     * Gets the archive Uri from the given network id
+     *
+     * @param gateway
+     */
+    private static getArchiveUri(gateway: IGateway | IGatewayPool): Promise<ContentUri> {
+        if (gateway.getArchiveUri()) {
+            return Promise.resolve(new ContentUri(gateway.getArchiveUri()))
+        }
+
+        return gateway.getEnsPublicResolverInstance().then(async instance => {
+            let entityEnsNode: string
+            switch (gateway.getNetwork()) {
+                case "mainnet":
+                    entityEnsNode = keccak256(VOCDONI_MAINNET_ENTITY_ID)
+                    break
+                case "goerli":
+                    entityEnsNode = keccak256(VOCDONI_GOERLI_ENTITY_ID)
+                    break
+                case "rinkeby":
+                    entityEnsNode = keccak256(VOCDONI_RINKEBY_ENTITY_ID)
+                    break
+                case "xdai":
+                    if (gateway.getEnvironment() === "prod") {
+                        entityEnsNode = keccak256(VOCDONI_XDAI_ENTITY_ID)
+                        break
+                    }
+                    entityEnsNode = keccak256(VOCDONI_XDAI_STG_ENTITY_ID)
+                    break
+                case "sokol":
+                    entityEnsNode = keccak256(VOCDONI_SOKOL_ENTITY_ID)
+                    break
+            }
+            return instance.text(entityEnsNode, TextRecordKeys.VOCDONI_ARCHIVE)
+        }).then((uri: string) => {
+            if (!uri) {
+                throw new Error()
```

Comment:
Some message? Some custom Error class?

Reply:
The intended behaviour here is to trigger the catch function of the caller, which is `getProcessFromArchive` (or `getProcess` after the renaming of the namespace), which triggers the error from the original caller.